### PR TITLE
Create Deployment instead of Pod (credentials tasks) #8182

### LINKED ISSUE #8182: Create Deployment instead of Pod (credentials tasks)
**This is a...** 

- [x] Feature Request
- [ ] Bug Report

**Problem:**
Several Task topics have the reader create a Pod. It would be better to have the reader create a Deployment.

**Proposed Solution:**
For example, take this topic: [Distribute Credentials Securely Using Secrets](https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/).
One of the YAML file has this Pod manifest:

```
apiVersion: v1
kind: Pod
metadata:
  name: secret-test-pod
spec:
  containers:
    - name: test-container
      image: nginx
      volumeMounts:
          # name must match the volume name below
          - name: secret-volume
            mountPath: /etc/secret-volume
  # The secret data is exposed to Containers in the Pod through a Volume.
  volumes:
    - name: secret-volume
      secret:
        secretName: test-secret
```

Change the YAML file to a Deployment manifest like this:

```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: secret-test
spec: # DeploymentSpec
  selector:
    matchLabels:
      purpose: test-secret-access
  replicas: 2
  template: # PodTemplateSpec
    metadata:
      labels:
        purpose: test-secret-access
    spec: # PodSpec
      containers:
        - name: test-container
          image: nginx
          volumeMounts:
              # name must match the volume name below
              - name: secret-volume
                mountPath: /etc/secret-volume
      # The secret data is exposed to Containers in the Pod through a Volume.
      volumes:
        - name: secret-volume
          secret:
            secretName: test-secret
```

Adjust the the text before and after any changed YAML files.

For example, change "Here is a configuration file you can use to create a Pod:" to "Here is a configuration file you can use to create a Deployment:".

**Page to Update:**
https://kubernetes.io/docs/tasks/inject-data-application/distribute-credentials-secure/
https://kubernetes.io/docs/tasks/inject-data-application/podpreset/

### PR DESCRIPTION
**Pull request by jeremy-novey**
/assign @bradtopol
/assign @steveperry-53

Fixes #8182

### CODE REVIEW

**Comment by jeremy-novey** [comment]
cla/linuxfoundation — jeremy-novey needs CLA - COMPLETED

**Comment by zacharysarah** [comment]
@jeremy-novey 👋+1 to assigning a value to `purpose:`. 

@tengqm, feel free to `/lgtm` when you're satisfied.

/approve

**Comment by steveperry-53** [comment]
/hold
I'm putting a hold on this, because I'm in the process of testing the revised steps.

File: content/en/docs/tasks/inject-data-application/distribute-credentials-secure.md [hunk 1]
```diff
@@ -1,174 +1,174 @@
...
+       kubectl get deployment secret-test
+
+    Output:
+
+        NAME               DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+        secret-test        2         2         2            2           12s
+
+1. Get a shell into the Container that is running in your Deployment:
+
+       kubectl exec -it secret-test -- /bin/bash
```
Comment (steveperry-53) [inline content/en/docs/tasks/inject-data-application/distribute-credentials-secure.md]: There is no longer a Pod named secrete-test, so we'll have to change this command to something like `kubectl exec -it secret-test-6fdd57d45-q9wj5 -- /bin/bash`. Stand by, and I'll come up with a concrete suggestion.

Comment (steveperry-53) [reply content/en/docs/tasks/inject-data-application/distribute-credentials-secure.md]: Here's what I think we should do:

`kubectl get pods`

The output shows the names of the Pods in your Deployment:

```
NAME                          READY     STATUS    RESTARTS   AGE
secret-test-6fdd57d45-q9wj5   1/1       Running   0          1m
secret-test-6fdd57d45-wtjtv   1/1       Running   0          1m
```

Get a shell into the Container that is running in one of your Pods:

`kubectl exec -it <pod-name> -- /bin/bash`

where `<pod-name>` is the name of one of your Pods.

File: content/en/docs/tasks/inject-data-application/distribute-credentials-secure.md [hunk 2]
```diff
@@ -1,174 +1,174 @@
...
+       kubectl get deployment secret-envars-test
+
+    Output:
+
+        NAME               DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+        secret-test        2         2         2            2           12s
+		
+1. Get a shell into the Container that is running in your Deployment:
+
+       kubectl exec -it secret-envars-test -- /bin/bash
```
Comment (steveperry-53) [inline content/en/docs/tasks/inject-data-application/distribute-credentials-secure.md]: See previous comment about getting a Pod name before `kubectl exec ...`.

**Comment by steveperry-53** [comment]
@jeremy-novey, Thanks for doing this. A couple of comments in line.

File: content/en/docs/tasks/inject-data-application/podpreset.md [hunk 3]
```diff
@@ -1,163 +1,164 @@
...
+```
+
+List the running Deployments:
+
+       kubectl get deployments
+
+    Output:
+
+        NAME               DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+        secret-test        2         2         2            2           12s
```
Comment (steveperry-53) [inline content/en/docs/tasks/inject-data-application/podpreset.md]: I don't think this Output is correct. The Deployment is not named secret-test.

**Comment by zacharysarah** [comment]
@jeremy-novey 👋 Please respond to feedback and resolve the merge conflicts in order to continue.

**Comment by chenopis** [comment]
@jeremy-novey ping

**Comment by zacharysarah** [comment]
@jeremy-novey Please feel free to reopen this PR when you're ready.

**Pull request closed without merging**